packages/p2p/src/pages/my-ads/buy-ad-payment-methods-list/buy-ad-payment-methods-list.tsx

```diff
@@ -93,63 +116,84 @@ const BuyAdPaymentMethodsList = ({
         }
     };
 
-    const onClickPaymentMethodItem = value => {
-        if (value && !my_ads_store.payment_method_names.includes(value)) {
-            if (my_ads_store.current_method.is_deleted) {
-                onEditPaymentMethodItem(value, my_ads_store.current_method.key);
-                my_ads_store.setCurrentMethod({ ...my_ads_store.current_method, key: null, is_deleted: false });
+    const onClickPaymentMethodItem = (value: string) => {
+        if (value && !payment_method_names.includes(value)) {
+            if (current_method.is_deleted) {
+                onEditPaymentMethodItem(value, current_method.key);
+                setCurrentMethod({ ...current_method, key: null, is_deleted: false });
                 setShowList(false);
                 setHideList(true);
-            } else if (my_ads_store.payment_method_names.length < MAX_PAYMENT_METHOD_SELECTION) {
-                my_ads_store.payment_method_names.push(value);
+            } else if (payment_method_names.length < MAX_PAYMENT_METHOD_SELECTION) {
+                payment_method_names.push(value);
                 setSelectedMethods([...selected_methods, value]);
                 setPaymentMethodsList(payment_methods_list.filter(payment_method => payment_method.value !== value));
             }
             if (typeof touched === 'function') touched(true);
         }
     };
 
-    const onClickIcon = (payment_method, key, setFieldValue) => {
+    const onClickIcon = (
+        payment_method: string,
+        key: number,
+        setFieldValue: (value: string, method: string) => void
+    ) => {
         if (close_icon) {
             onEditPaymentMethodItem(payment_method, key);
             setShowList(true);
             setFieldValue('payment_method', '');
-        } else if (!close_icon && !my_ads_store.current_method.is_deleted) {
+        } else if (!close_icon && !current_method.is_deleted) {
             onClickDeletePaymentMethodItem(payment_method);
         }
     };
 
+    const getComponent = () => (
```

Comment:
can we make the name to be more specific 
```suggestion
    const getPaymentMethodHint = () => (
```